
\section{Existing methods in drug repurposing and causal analysis}

Resources such as the Comparative Toxicogenomics Database, Connectivity Map (CMap) ~\cite{lamb2007connectivity, lamb2006connectivity}, or Library of Integrated Network-based Cellular Signatures (LINCS), capture collective existing knowledge about the genes that are affected by a multitude of  chemicals, toxicants or drugs. This type of knowledge can be used for many purposes which in turn can generally be categorized into two main directions: drug repositioning and causal analysis.

\subsection{Drug repositioning}

The goal of drug repositioning is to identify new therapeutic applications for existing drugs. Since these drugs are already approved, they can skip the Phase I clinical trials in the drug approval pipeline, i.e. testing the safety of the drug. Therefore this approach is faster and more cost-efficient than the process of new drug discovery, which takes on average 15 years and more than one billion dollars for each drug \cite{chong2007new}. Most often the gene profiling of control samples and treated samples are compared to obtain a list of DE genes, also defined in some literatures as the summary compound's effect \cite{shaw2003transcriptional}. 
%In~\cite{di2005chemogenomic}, Bernando \textit{et al.} proposed an engineered reversed model of the compound-exposed cells' gene regulatory network to compute the likelihood of the gene products and associated pathways targeted by that compound. 
There are two main approaches of computational and knowledge-based drug repositioning methods: these summary compound's effect are compared to a disease-associated DE genes which obtained by the contrast between healthy and disease samples' gene expression (disease-based approaches); or to other compound's effect (drug-based approaches)~\cite{iorio2013transcriptional}. 

The disease-based approaches hypothesize that if the compound's effect are negatively correlated with the disease DE genes, e.g. a gene is up-regulated by a drug's effect and down-regulated by a disease, that compound would be a good candidate to revert the phenotype's DE genes, and hence can potentially suppress the phenotype~\cite{sirota2011discovery, mcart2011identification}. Some examples of studies using this idea in drug repurposing: Claerhout \textit{et al.} proposed using vorinostat as a candidate treatment for gastric cancer~\cite{claerhout2011gene}; Chen \textit{et al.} successfully identified and verified chlorpromazine and trifluoperazine as the alternative for sorafenib to treat hepatocellular carcinoma~\cite{chen2011gene}; Dudley \textit{et al.} proposed topiramate which was approved for epilepsy as a alternative treatment for inflammatory bowel disease~\cite{dudley2011computational}. 

The drug-based methods work under an assumption that if two drugs evoke similar summary compound's effects, they could share a common mode of action~\cite{iorio2010identification, iorio2010discovery, wolpaw2011modulatory, wolpaw2011modulatory, hu2009human, chiang2009systematic}. 

The majority of the methods in both groups utilize Connectivity Map (cMap)~\cite{lamb2007connectivity, lamb2006connectivity} or The Library of Integrated Cellular Signatures (LINCS) as reference of signature of differential gene expressions of disease and drug responses. CMap consists 564 gene expression profiles obtained by exposing five human cell lines (MCF7, PC3, SKMEL5, HL60, and ssMCF7) to 164 distinct small-molecule perturbagens, selected to represent a broad range of activities, and including U.S. Food and Drug Administration (FDA)–approved drugs and non-drug bioactive “tool” compounds~\cite{lamb2006connectivity}. LINCS, a project funded by NIH, has used L1000 technology, which is a technology developed by cMap team, to generate over one million gene expressions profiles. LINCS is basically a matrix of 22,268 rows represents genes and over millions of columns correspond to millions of perturbations induced by small molecules. It is available via the LINCS web based application (http://www.lincscloud.org/l1000/). The advent of cMap and LINCS has paved the way for systematic large-scale research in drug repositioning.  Peyvandipour \textit{et al.} used the combination of cMap and LINCS together with Kyoto Encyclopedia of Genes Genomics (KEGG, \cite{ogata1999kegg}), a signaling pathway repository, to create a so-called drug-desease-network\cite{peyvandipour2018novel}. They later use the impact analysis ~\cite{draghici2007systems} to derive a score to each drug-disease pair, which is used for identifying candidate drugs.

\subsection{Causal analysis}

Causal analysis refers to an analysis that aims to infer the CDT that potentially causes the observed expression changes. The methods in this category often hypothesize that a drug compound could cause a disease phenotype if the compound's gene signature is positively correlated with disease's gene signature~\cite{huang2013inferring}. Although this approach uses the gene expression profiles to reach the same goal as our proposed method, it utilizes a totally different technique. A more direct approach to identify the CDT is applying graph theories on the cause-effect network between CDT and genes. Chindelevitch \textit{et al.} used two commercial knowledge bases, Selventa Inc. (\href{http://www.selventa.com}{http://www.selventa.com}) and Ingenuity Inc. (\href{http://www.ingenuity.com}{http://www.ingenuity.com}) to construct a network of molecular causal interactions  that would suggest molecular hypotheses that explain the observed changes in gene expression profiles. For each molecule, they used a scoring system which performs a subtraction of  the number of genes against the hypotheses from the number of genes supporting the hypotheses~\cite{chindelevitch2012causal}. Subsequently, they applied the distribution of the scores under the null and Fisher's exact test to compute the statistical significance. More recently, Kr{\"a}mer \textit{et al.} published a paper that presents the causal analysis approach in Ingenuity Pathway Analysis (IPA). This work has very similar goals with our approach, hence we will discuss and compare its performance with ours in the following subsections. Although there are computational methods using the similar techniques on specific applications, %e.g. ~\cite{pollard2005computational} on Type 2 Diabetes Mellitus, 
large-scale and more general attempts are scarce in this field.




