\section{Introduction}
\label{chap:Introduction}

%COVID-19 is one such infectious disease that emerged in 2020.
The COVID-19 pandemic, caused by the severe acute respiratory syndrome coronavirus 2 (SARS-CoV-2), emerged in late 2019 and rapidly evolved into one of the most significant global health crises of the 21st century. Originated in Wuhan, the virus spread across borders, prompting The World Health Organization (WHO) to declare it a Public Health Emergency of International Concern on January 30, 2020, and subsequently to recognize it as a pandemic on March 11, 2020. 
COVID-19 primarily spreads through respiratory droplets and aerosols, leading to symptoms ranging from mild respiratory issues to severe pneumonia and death. 
Characterized by its high transmissibility and severe respiratory impact, COVID-19 disrupted economies, healthcare systems, and daily life worldwide, claiming millions of lives and exposing vulnerabilities in global preparedness.
%The pandemic has had profound global impacts, causing widespread morbidity and mortality, straining healthcare systems, and disrupting economies and daily life worldwide.

Efforts to combat COVID-19 encompass multiple critical areas aimed at controlling the virus, treating infected individuals, and preventing further spread. Antiviral treatments are being developed and repurposed to inhibit viral replication and reduce disease severity. Vaccine development has been a major focus, leading to the rapid creation and distribution of vaccines like those based on mRNA and viral vector technologies (AstraZeneca, Johnson \& Johnson), which have significantly reduced severe cases and mortality. Diagnostic testing has played a crucial role in identifying infections, with molecular tests (PCR), rapid antigen tests, and serological assays enabling early detection and epidemiological tracking. Additionally, patient-supporting interventions such as oxygen therapy, and ventilatory support, have improved patient outcomes and reduced complications. Together, these efforts represent a comprehensive approach to managing the COVID-19 pandemic and mitigating its global impact.

Recent research identified a sub-group with severe COVID-19 patients developed a cytokine storm syndrome that might be responsible to their high mortality~\cite{mehta2020covid}. This hyper-inflammatory response, characterized by excessive cytokine production, can lead to severe complications such as multi-organ failure and acute respiratory distress syndrome (ARDS). Therefore, early identification followed by timely intervention can help mitigate the excessive immune response and improve survival rates. A proactive and personalized approach to managing cytokine storm is crucial for reducing mortality and enhancing patient recovery.

To understand the immune reactions triggered by SARS-CoV-2 and understand the complications seen in COVID-19 patients, many life science experiments focus on comparisons between COVID-19 samples (or SARS-CoV-2 infected cell lines) versus healthy samples.
Microarrays and more recently, RNASeq assays, allow researchers to measure all genes and subsequently  yield a list of differentially expressed (DE) genes.
The challenge is to translate these measurements and lists of DE genes into a better understanding of the underlying biological phenomena, and in particular an understanding of its mechanisms. Analysis approaches such as pathway analysis~\cite{DraghiciOntologicalToolsReview:2005,Khatri:2012, mitrea2013methods, tarca2013comparison, nguyen2018network, ihnatova2018critical, nguyen2019identifying}, network analysis~\cite{mitra2013integrative} and gene ontology (GO) analysis~\cite{DraghiciOntologicalToolsReview:2005,Rhee:2008}, have been very successful in the past two decades in helping to translate such lists of DE genes into meaningful insights of the underlying biological phenomena. 

Pathway databases such as Kyoto Encyclopedia of Genes and Genomes (KEGG)~\cite{Kanehisa:2000}, Reactome~\cite{croft2014reactome}, BioCarta~\cite{BioCartaWWW}, NCI-PID ~\cite{Schaefer:2009}, WikiPathways~\cite{pico2008wikipathways}, and PANTHER~\cite{thomas2003panther} model signaling pathways as networks in which nodes represent related genes or gene products and edges symbolize interactions among them based on prior knowledge. 
Pathway analysis approaches use available pathway databases and the given gene expression data to identify the pathways which are significantly impacted in a given condition. 
Because of the importance of this type of analysis, more than hundreds of pathway analysis methods have been proposed thus far~\cite{DraghiciOntologicalToolsReview:2005,Khatri:2002,mitrea2013methods}.
It would now become a challenge for any researcher to choose which pathway analysis method is the most suitable for their analysis. 
Although there have been some review studies before, all of them have some limitations.

Here, for the first time, we present a comparison of the performances of 13 representative pathways analysis methods on 88 real data sets from two species: human and mouse. 
To our knowledge, this is the highest number of real data sets used in a comparative study on pathway analysis methods.
The second assessment investigates the potential bias of each method and pathway.

This work provides precise, objective, and reproducible answers to the following important and currently unanswered questions: (i) is there any difference in performance between non-TB and TB methods?, (ii) is there a method that is consistently better than the others in terms of its ability to identify target pathways, accuracy, sensitivity, specificity, and the Area Under the receiver operating characteristic Curve (AUC)?, (iii) are there any specific pathways that are biased (in the sense of being more likely or less likely to be significant across all methods)?, and (iv) do specific methods have a bias toward specific pathways (e.g. is pathway X likely to be always reported as significant by method Y)?. 
This article provides a guidance to help researchers select the right method to deploy in analyzing their data based on any kind of scientific criteria. 
At the same time, this article will be of interest to any computational biologists or bioinformaticians involved in developing new analysis methods. 
For such researchers, this article is expected to become the benchmark against which any future analysis method will have to be compared. 
Finally, because of the bias analysis of all known KEGG pathways included here, this article is also expected to be extremely useful to many people involved in the curation and creation of pathway databases.

More importantly, the emergence of COVID-19 highlighted the urgent need for a swift drug discovery process to treat affected patients. %Drug repurposing presents a promising solution to this challenge.
The traditional drug discovery and approval process is time consuming, expensive, and high-risk endeavor, often taking 10–15 years and costing billions of dollars to bring a single new drug to market. From initial discovery to preclinical studies, followed by rigorous clinical trials and regulatory approvals, this process is fraught with challenges, including high failure rates and substantial financial burdens~\cite{adams2006estimating, dickson2009cost, dimasi2003price}.
Given these obstacles, drug repurposing—the strategy of identifying new therapeutic uses for existing drugs—has emerged as a cost-effective and time-efficient alternative. By utilizing drugs that have already been tested for safety in humans, this approach can significantly reduce development timelines, lower research costs, and expedite regulatory approval, making it a valuable approach for addressing urgent medical needs, such as emerging infectious diseases and rare disorders. Drug repurposing not only maximizes the utility of existing pharmaceuticals but also enhances the efficiency of drug development in an era where rapid therapeutic innovation is critical~\cite{chong2007new, ashburn2004drug}.

One popular approach to discover the potential drug that treat the condition is to scan the existing drug database to identify the ones that can reverse the list of observed DE genes.  
This approach, in turn, falls under an umbrella of bigger problem in this field that focuses on the identification of upstream regulators that may explain the observed changes. In principle, such upstream regulators could be of different types including: genes (e.g. gene encoding transcription factors), miRNA, drugs, chemicals, or toxicants. This type of analysis is generally referred to as ``causal analysis''~\cite{schadt:2005, chindelevitch2012causal, kramer2013causal, felciano2013predictive}. 

Some diseases are caused by exposing to harmful or toxic environment, such as wild fire smoke or polluted water sources. These toxicants can be either a product of human activities or from natural disasters. In these cases, the presence of a chemical substance  is responsible for the changes in  the gene expression profiles and  therefore, for creating the phenotypes. 
In other situations,  a phenotype and its associated  gene expression changes  can be caused by the lack of a necessary chemical  that plays an important metabolic role, e.g. iodine deficiency. 
Identifying the chemical, drug, or toxicant (CDT) that perturbs the patients' gene expression level is a crucial step to pinpoint the cause and therefore help finding a suitable treatment for the patients.


Because understanding the effects of various CDTs is so important, the associations between chemicals and gene products have been studied intensely in the past decade and are available in several curated public chemical knowledge bases, such as the Comparative Toxicogenomics Database ~\cite{mattingly2006comparative}, KEGG~\cite{Kanehisa:2000},  DrugBank~\cite{law2014drugbank}, KEGG BRITE~\cite{kanehisa2006genomics}, BRENDA~\cite{schomburg2004brenda}, SuperTarget~\cite{gunther2007supertarget}, etc. 


Some notable examples of research in this fields are: Schadt \emph{et al.} successfully identified three new genes in susceptibility to obesity by using this approach~\cite{schadt:2005};
Pollard \emph{et al.} proposed a computational model to define the molecular causes of Type 2 Diabetes Mellitus~\cite{pollard2005computational};  A Pfizer research group created networks of molecular causal interactions by integrating available biological knowledge, mainly from two commercial vendors: Selventa Inc. (\href{http://www.selventa.com}{http://www.selventa.com}) and Ingenuity Inc. (\href{http://www.ingenuity.com}{http://www.ingenuity.com})~\cite{chindelevitch2012causal}.

%In the next part of the study, we propose another similar approach to identify upstream regulators as an attempt to search for an alternative treatments for COVID-19, that has caused around 800 millions cases and around 7 millions death (as of December 2023). 
%
%Most current efforts related to COVID-19 span a number of areas as follows: i) antivirals, ii) vaccine development, iii) diagnostic tests, and iv) patient-supporting interventions. 
%Without reducing the significance and impact of any of the areas above, there is an important aspect that has not been elucidated: the identification and treatment of patients developing critical conditions and risk of mortality.  Recently, Mehta \textit{et al.} stated that ``Accumulating evidence suggests that a subgroup of patients with severe COVID-19 might have a cytokine storm syndrome'' that correlates with high mortality~\cite{mehta2020covid}.
%Therefore, identification and appropriate management of the patients developing cytokine storm syndrome is critical for successful outcomes. Treatment of hyper-inflammation in these patients using existing, approved therapies with proven safety profiles could address the immediate need to reduce the rising mortality. 
%
%COVID-19 has several distinct clinical phases:  an infection phase, a viral replication phase, an inflammatory phase, and in some patients, a hyper-inflammatory phase or cytokine storm~\cite{siddiqi2020covid,Ayres2020:survivingCOVID19}. After the initial viral phase of the illness, some patients will develop a cytokine storm which has being associated with the acute respiratory distress syndrome (ARDS) and mortality.  Therefore, in order to decrease the risk of mortality is necessary to distinguish the phase where the viral pathogenicity is dominant versus when the host inflammatory response overtakes the pathology~\cite{siddiqi2020covid,Ayres2020:survivingCOVID19}. A potential approach is to develop interventions that could inhibit/prevent the hyper-inflammatory process leading to the cytokine storm.
%A strong argument in favor of also targeting the host response is offered by the data on influenza. Even though influenza patients receive optimal anti-viral therapy, approximately 25\% of the critically ill influenza patients still die~\cite{Ayres2020:survivingCOVID19,louie2012treatment}. This suggests that anti-viral therapy alone will not be sufficient for COVID-19 either, and the host response to the virus still needs to be taken into consideration.
%
%However, approaches aiming at modulating the immune response face some concerns. In particular, it may seem counter-intuitive to try to diminish the immune response in a patient whose immune system is fighting against a virus. Modulating the immune system is likely unnecessary and counter-productive for patients whose immune system is doing a good job at resolving the infection, while it could potentially be life-saving for those whose inflammatory response has become dysregulated. If a patient has developed severe respiratory symptoms and is hypoxic, the host response that lead to ARDS, sepsis, and organ failure has already been initiated~\cite{mehta2020covid}. At this point, the focus should shift to supporting the patient's systems and preventing collapse triggered by hyper-inflammation~\cite{Ayres2020:survivingCOVID19}. 
%
%In order to identify the best potential therapeutic approach, we performed a transcriptome analysis of tissues and cell samples infected with SARS-CoV-2 in order to understand the main mediators of the inflammatory process. Once characterized the inflammatory pathways we identified drugs that would mitigate or alleviate some of the devastating over-reactions of the host's immune system (e.g. cytokine storm). Finally, we evaluated the efficacy of the identified drug in a small cohort of COVID-19 patients.

%Given a gene expression profile, a pathway analysis method provide the insight of the immune reaction triggered by SARS-CoV-2 and a causal analysis method would discover from the list of existing drugs candidates that could treat severe COVID-19 patients.
%
%In this thesis, we would first analyze the current situation of pathway analysis field and provide a guidance for  researchers to select the right pathway analysis method based on each individual's research criteria. In the second part, we proposed an upstream analysis method that can be used for drug repurposing. 

In this thesis, we would first provide a comprehensive review of one of the most popular pathway analysis methods so we can choose a suitable method for our COVID-19 study. In the second part, we proposed an upstream analysis method that can be used for drug repurposing to identify at least one potential drug to treat severe COVID-19 patients. 
%Subsequently, we would apply the best pathway analysis approach recommended and the proposed upstream analysis to study the real data sets of severe cases of COVID-19 patients. 
Using pathway analysis, we would identify that these patients were developing a cytokine storm syndrome, which is associated with high mortality rate in COVID-19 patients. Treatment of hyper-inflammation in these patients using existing, approved therapies with proven safety profiles could address the immediate need to reduce mortality. 
Then, we would apply the drug repurposing approach to propose a possible treatment for COVID-19 patients with a hyper-inflammatory phase. 
We analyzed  the changes in the gene expression, pathways and putative mechanisms induced by SARS-CoV2 in  NHBE, and A549 cells, as well as COVID-19 lung vs. their respective controls. We used these  changes to identify FDA approved drugs that could be repurposed to help COVID-19 patients with severe symptoms related to hyper-inflammation. 

The thesis will be organized as follows:

\textbf{Chapter 2}  outlines the contemporary problems within the field that serve as the focal point for our contributions. This chapter also delves into an extensive review of existing methodologies and related research pertinent to this thesis.
\textbf{Chapter 3} highlights the limitations as well as the challenges of the existing methods mentioned in previous chapter.
Moving forward to \textbf{chapter 4}, we describe our proposal a framework for benchmarking pathway analysis methods, that provide guidance for researchers to choose the most suitable method for their goals. \textbf{Chapter 5} marks the introduction of our innovative approach, wherein we provide a comprehensive elucidation of the methodology. In the second part of the chapter 5, we elucidate the validation and assessment procedures devised to evaluate the efficacy of our proposed method including intricate details on testing hypotheses, dataset selection, competitive methods and the criteria for declaring success. We also discuss how our proposed method would improve and address the limitations of the existing methods.
In \textbf{chapter 6}, we analyze gene expression data sets of COVID-19 patients to propose an alternative treatment using the aforementioned approach.
The last chapter, \textbf{chapter 7} summarizes the thesis and lay out the work for the final thesis.

%\textbf{Chapter 6} summarizes the proposed approaches as well as the plan for the future work.}

%Chapter 2 proposes a novel approach to identify the upstream CDT regulator. 
%We describe the CDT-drug association knowledge base and data sets used in the study as well as the hypotheses tested. Subsequently, we describe the method to compute the statistical significance. Next, we define the criteria to assess the performance and compare with other benchmarking methods. We discuss the possible limitations of the method at the end of the chapter.
%
%In Chapter 3, we describe our approach to identify potential alternative drugs for treatment of severe  COVID-19 cases. Subsequently, we introduce the data sets and experiments used in this study. We present some preliminary results in this section.
%
%Finally, chapter 4 describes the future development plans that includes the validation and assessment for the approaches in chapter 2 and chapter 3. 



